### PR TITLE
Reorganize the way we send info to lesson edit

### PR DESCRIPTION
**Pull request by dmcavoy**
For the lesson editor we were sending a bunch of data down through `lesson.summarize_for_lesson_edit` which was not getting edited and not about the lesson itself. Since all the information was getting sent in one big object it was making it harder to pass it around appropriately on the front end such as in redux. This PR takes the data that was in `summarize_for_lesson_edit` on lesson and splits it out into more logical groupings. 

One grouping is to add a `summarize_for_lesson_edit` on Script so that we can get the necessary information about the script. None of that information is updated during the editing of the lesson edit page so this draws a nice separation. 

The other grouping is the information we need to search for levels. That is now its own object as well. 

This allows us to have 3 things in the activitiesEditorRedux which we are exporting: activities, unitInfo, and levelSearchingInfo.

This sets us up to add more data about the unit in https://github.com/code-dot-org/code-dot-org/pull/43214

## Links

Done in response to feedback on https://github.com/code-dot-org/code-dot-org/pull/43214

## Testing story

- Existing tests and updates to them

### CODE REVIEW

**Review by davidsbailey — APPROVED** [review]
great design choice, Dani! I really like how the lesson data is kept separate from the other data, since it's nice to be able to send just this data back to the server on update. thanks for taking initiative here!

File: dashboard/app/models/script.rb [hunk 1]
```diff
@@ -1665,6 +1665,15 @@ def summarize_for_unit_edit
     summary
   end
 
+  def summarize_for_lesson_edit
+    {
+      isLaunched: script.launched?,
```
Comment (davidsbailey) [inline dashboard/app/models/script.rb]: I tried to rebase off of this PR and i think I am getting an error here because it can't find `script`

Comment (dmcavoy) [reply dashboard/app/models/script.rb]: Good catch. Just updated

File: apps/src/lib/levelbuilder/lesson-editor/activitiesEditorRedux.js [hunk 2]
```diff
@@ -33,17 +34,14 @@ export const NEW_LEVEL_ID = '-1';
...
-  searchOptions,
-  programmingEnvironments,
-  lessonExtrasAvailableForUnit
-) => ({
-  type: INIT,
-  activities,
-  searchOptions,
-  programmingEnvironments,
-  lessonExtrasAvailableForUnit
+export const initg = levelSearchingInfo => ({
```
Comment (dmcavoy) [inline apps/src/lib/levelbuilder/lesson-editor/activitiesEditorRedux.js]: fix typo

File: dashboard/app/models/script.rb (hunk 1 above)
Comment (dmcavoy) [reply dashboard/app/models/script.rb]: For some reason its not getScriptData is not working for the new data I added. I'm trying to dig in and figure out why

Comment (dmcavoy) [reply dashboard/app/models/script.rb]: Found the issue! Should be working now

**Pull request merged**